panic: runtime error: index out of range in LogRoller

I have this extra file: `/tmp/cups-connector.txt` which I was using to store a diff.

When I run gcp-cups-connector I get:

```
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/google/cups-connector/log.(*LogRoller).roll(0xc820074c80, 0x0, 0x0)
        /home/agoode/gocode/src/github.com/google/cups-connector/log/logroller.go:131 +0xa84
github.com/google/cups-connector/log.(*LogRoller).openFile(0xc820074c80, 0x0, 0x0)
        /home/agoode/gocode/src/github.com/google/cups-connector/log/logroller.go:83 +0x31
github.com/google/cups-connector/log.(*LogRoller).Write(0xc820074c80, 0xc82001a240, 0x53, 0x60, 0x0, 0x0, 0x0)
        /home/agoode/gocode/src/github.com/google/cups-connector/log/logroller.go:62 +0xb1
fmt.Fprintf(0x7f3076a729e8, 0xc820074c80, 0x8ce380, 0xb, 0xc8200b9208, 0x3, 0x3, 0x0, 0x0, 0x0)
        /usr/lib/golang/src/fmt/print.go:189 +0xb2
github.com/google/cups-connector/log.log(0x417401, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc8200b96b0, 0x1, 0x1)
        /home/agoode/gocode/src/github.com/google/cups-connector/log/log.go:141 +0x1132
github.com/google/cups-connector/log.Error(0xc8200b96b0, 0x1, 0x1)
        /home/agoode/gocode/src/github.com/google/cups-connector/log/log.go:166 +0x5e
main.connector(0xc8200f4d80, 0x0)
        /home/agoode/gocode/src/github.com/google/cups-connector/gcp-cups-connector/gcp-cups-connector.go:93 +0x6e7
main.main.func1(0xc8200f4d80)
        /home/agoode/gocode/src/github.com/google/cups-connector/gcp-cups-connector/gcp-cups-connector.go:45 +0x21
github.com/codegangsta/cli.(*App).Run(0xc8200f4c60, 0xc82000a2d0, 0x1, 0x1, 0x0, 0x0)
        /home/agoode/gocode/src/github.com/codegangsta/cli/app.go:180 +0x1028
github.com/codegangsta/cli.(*App).RunAndExitOnError(0xc8200f4c60)
        /home/agoode/gocode/src/github.com/codegangsta/cli/app.go:186 +0x54
main.main()
        /home/agoode/gocode/src/github.com/google/cups-connector/gcp-cups-connector/gcp-cups-connector.go:47 +0x20d

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /usr/lib/golang/src/runtime/asm_amd64.s:1696 +0x1

goroutine 5 [syscall]:
os/signal.loop()
        /usr/lib/golang/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
        /usr/lib/golang/src/os/signal/signal_unix.go:28 +0x37
```


config file:
```
{
  "xmpp_server": "talk.google.com",
  "xmpp_port": 443,
  "gcp_xmpp_ping_timeout": "5s",
  "gcp_xmpp_ping_interval_default": "2m",
  "gcp_base_url": "https://www.google.com/cloudprint/",
  "gcp_oauth_client_id": "539833558011-35iq8btpgas80nrs3o7mv99hm95d4dv6.apps.googleusercontent.com",
  "gcp_oauth_client_secret": "V9BfPOvdiYuw12hDx5Y5nR0a",
  "gcp_oauth_auth_url": "https://accounts.google.com/o/oauth2/auth",
  "gcp_oauth_token_url": "https://accounts.google.com/o/oauth2/token",
  "gcp_max_concurrent_downloads": 5,
  "cups_max_connections": 50,
  "cups_connect_timeout": "5s",
  "cups_job_queue_size": 3,
  "cups_printer_poll_interval": "1m",
  "cups_printer_attributes": [
    "cups-version",
    "device-uri",
    "document-format-supported",
    "print-color-mode-default",
    "print-color-mode-supported",
    "printer-name",
    "printer-info",
    "printer-location",
    "printer-make-and-model",
    "printer-state",
    "printer-state-reasons",
    "printer-uuid",
    "marker-names",
    "marker-types",
    "marker-levels",
    "copies-default",
    "copies-supported",
    "number-up-default",
    "number-up-supported",
    "orientation-requested-default",
    "orientation-requested-supported",
    "pdf-versions-supported"
  ],
  "cups_job_full_username": false,
  "cups_ignore_raw_printers": true,
  "copy_printer_info_to_display_name": true,
  "prefix_job_id_to_job_title": false,
  "display_name_prefix": "",
  "monitor_socket_filename": "/tmp/cups-connector-monitor.sock",
  "snmp_enable": false,
  "snmp_community": "public",
  "snmp_max_connections": 100,
  "local_printing_enable": true,
  "cloud_printing_enable": false,
  "log_file_name": "/tmp/cups-connector",
  "log_file_max_megabytes": 1,
  "log_max_files": 3,
  "log_level": "INFO",
  "log_to_journal": false
}
```

This is a local denial of service, since any user can write to /tmp.

--- a/log/logroller.go
+++ b/log/logroller.go
@@ -120,17 +120,21 @@ func (lr *LogRoller) roll() error {
 	}
 
 	// Get all rolled logs, plus some.
-	matches, err := filepath.Glob(lr.fileName + ".*")
+	allFiles, err := filepath.Glob(lr.fileName + ".*")
 	if err != nil {
 		return err
 	}
 
 	// Get number suffixes from the rolled logs; ignore non-matches.
-	numbers := make(sortableNumberStrings, 0, len(matches))
-	for _, match := range matches {
-		number := rollPattern.FindStringSubmatch(match[len(lr.fileName):])[1]
-		if _, err := strconv.ParseUint(number, 10, 16); err == nil {
-			numbers = append(numbers, number)
+	numbers := make(sortableNumberStrings, 0, len(allFiles))
+	for _, file := range allFiles {
+		match := rollPattern.FindStringSubmatch(file[len(lr.fileName):])
+		if len(match) < 2 {
+			continue
+		}
+		if _, err := strconv.ParseUint(match[1], 10, 16); err == nil {
+			// Keep the string form of the number.
+			numbers = append(numbers, match[1])
 		}
 	}
 

